source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,221 @@
+        return maskedPointCloud;
+    }
+
+    void visualizeDepthMap(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    {
+        // Converting to Depth map in OpenCV format
+        cv::Mat zColorMap = pointCloudToCvZ(pointCloud);
+        // Visualizing Depth map
+        cv::namedWindow("Depth map", cv::WINDOW_AUTOSIZE);
+        cv::imshow("Depth map", zColorMap);
+        cv::waitKey(0);
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB> convertToPCLPointCloud(const Zivid::PointCloud &pointCloud)
+    {
+        const auto data = pointCloud.copyData<Zivid::PointXYZColorRGBA>();
+
+        // Creating PCL point cloud structure
+        pcl::PointCloud<pcl::PointXYZRGB> pointCloudPCL;
+        pointCloudPCL.width = pointCloud.width();
+        pointCloudPCL.height = pointCloud.height();
+        pointCloudPCL.is_dense = false;
+        pointCloudPCL.points.resize(pointCloudPCL.width * pointCloudPCL.height);
+
+        // Filling in point cloud data
+        for(size_t i = 0; i < pointCloudPCL.points.size(); ++i)
+        {
+            pointCloudPCL.points[i].x = data(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].y = data(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].z = data(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].r = data(i).color.r; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].g = data(i).color.g; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloudPCL.points[i].b = data(i).color.b; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+        return pointCloudPCL;
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::string fileName = std::string(ZIVID_SAMPLE_DATA_DIR) + "/Zivid3D.zdf";
+        std::cout << "Reading ZDF frame from file: " << fileName << std::endl;
+        const auto frame = Zivid::Frame(fileName);
+
+        std::cout << "Getting point cloud from frame" << std::endl;
+        const auto pointCloud = frame.pointCloud();
+
+        const int pixelsToDisplay = 300;
+        std::cout << "Generating binary mask of central " << pixelsToDisplay << " x " << pixelsToDisplay << "pixels."
+                  << std::endl;
+        const int height = pointCloud.height();
+        const int width = pointCloud.width();
+        const int heightMin = (height - pixelsToDisplay) / 2;
+        const int heightMax = (height + pixelsToDisplay) / 2;
+        const int widthMin = (width - pixelsToDisplay) / 2;
+        const int widthMax = (width + pixelsToDisplay) / 2;
+        cv::Mat mask = cv::Mat::zeros(height, width, CV_8U);
+        cv::rectangle(mask,
+                      cv::Point(widthMin, heightMin),
+                      cv::Point(widthMax, heightMax),
+                      cv::Scalar(255, 255, 255),
+                      -1);
```

Comment:
Can we use a named value instead of `-1`? I see

> thickness | Thickness of lines that make up the rectangle. Negative values, like FILLED, mean that the function has to draw a filled rectangle.

So does `cv::FILLED` work instead?

Reply:
Yes, done